src/components/PostDetails/PostDetails.jsx

```diff
@@ -1,45 +1,84 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
 import { NewCommentForm } from '../NewCommentForm';
 import './PostDetails.scss';
 
-export const PostDetails = () => (
-  <div className="PostDetails">
-    <h2>Post details:</h2>
-
-    <section className="PostDetails__post">
-      <p>sunt aut facere repellat provident occaecati excepturi optio</p>
-    </section>
-
-    <section className="PostDetails__comments">
-      <button type="button" className="button">Hide 2 comments</button>
-
-      <ul className="PostDetails__list">
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>My first comment</p>
-        </li>
-
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>sad sds dfsadf asdf asdf</p>
-        </li>
-      </ul>
-    </section>
-
-    <section>
-      <div className="PostDetails__form-wrapper">
-        <NewCommentForm />
-      </div>
-    </section>
-  </div>
-);
+import { getPostDetails } from '../../api/posts';
+import { getPostComments } from '../../api/comments';
+import { Loader } from '../Loader';
+import { Comments } from '../Comments';
+
+export const PostDetails = ({ selectedPostId }) => {
+  const [post, getPost] = useState(null);
+  const [showedComments, showComments] = useState(true);
+  const [comments, getComments] = useState(null);
+
+  useEffect(() => {
+    getPostDetails(selectedPostId)
+      .then(getPost);
+
+    getPostComments(selectedPostId)
+      .then(getComments);
+  }, [selectedPostId]);
+
+  if (!post) {
+    return (
+      <Loader />
+    );
+  }
```

Comment:
use variable loading in your state for this, because in your code loader works only when you got a null inside the post, you can set variable loading as a true when you send a request and as false when you got a response.